projects/swimlane/ngx-ui/src/lib/components/drawer/drawer.component.ts

```diff
@@ -90,48 +93,18 @@ export class DrawerComponent implements OnDestroy {
   private _zIndex: number;
   private _closeOnOutsideClick: boolean;
 
+  ngOnInit() {
+    this.position = this.isRoot ? 'fixed' : 'absolute';
+    this.setDimensions(this.size);
+  }
+
   ngOnDestroy() {
     this.close.emit(true);
   }
 
   setDimensions(size: number): void {
-    const winWidth = window.innerWidth;
-    const winHeight = window.innerHeight;
-    let height: string;
-    let width: string;
-    let transform: string;
-
-    if (this.isLeft) {
-      if (size) {
-        const innerWidth = size;
-        const widthPercent = (innerWidth / 100) * winWidth;
-        const newWidth = Math.ceil(widthPercent);
-
-        height = '100%';
-        width = `${newWidth}px`;
-        transform = `translate(-${width}, 0px)`;
-      } else {
-        transform = 'translate(100%, 0)';
-      }
-    } else if (this.isBottom) {
-      if (size) {
-        const innerHeight = size;
-        const heightPercent = (innerHeight / 100) * winHeight;
-        const newHeight = Math.ceil(heightPercent);
-
-        width = '100%';
-        height = `${newHeight}px`;
-        transform = `translate(0px, -${height})`;
-      } else {
-        transform = 'translate(0, 100%)';
-      }
-    }
-
-    setTimeout(() => {
-      this.heightSize = height;
-      this.widthSize = width;
-      this.transform = transform;
-    });
```

Comment:
I removed this code because during testing, I noticed that this code that was calculating the transform was never updating the element's transform property. The transform set by the drawer animation always took precedence and would not change even after subsequent calls to `setDimensions`. I also noticed that the code was calculating exact px sizes, even though the size property of the element is in percentages relative to 100%. It worked in that way when the width or height were just set to the specified percentage value. So I was able to greatly simplify this code.